materialize-snowflake/config.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/iancoleman/orderedmap"
+	"github.com/invopop/jsonschema"
+	"regexp"
+	"strings"
+
+	m "github.com/estuary/connectors/go/protocols/materialize"
+	sf "github.com/snowflakedb/gosnowflake"
+)
+
+// config represents the endpoint configuration for snowflake.
+// It must match the one defined for the source specs (flow.yaml) in Rust.
+type config struct {
+	// TODO(mahdi): the Host and Account config is very confusing since Snowflake has multiple ways and very specific requirements
+	// for specifying the url of an instance. Ideally we should just accept the URL directly and use that, that would save us and
+	// users some headache
+	// See: https://docs.snowflake.com/en/user-guide/admin-account-identifier#non-vps-account-locator-formats-by-cloud-platform-and-region
+	Host      string `json:"host" jsonschema:"title=Host URL,description=The Snowflake Host used for the connection. Must include the account identifier and end in .snowflakecomputing.com. Example: orgname-accountname.snowflakecomputing.com (do not include the protocol)." jsonschema_extras:"order=0,pattern=^[^/:]+.snowflakecomputing.com$"`
+	Account   string `json:"account" jsonschema:"title=Account,description=The Snowflake account identifier." jsonschema_extras:"order=1"`
+	User      string `json:"user" jsonschema:"-"`
+	Password  string `json:"password" jsonschema:"-"`
```

Comment:
The default rendering of the form is:
<img width="1241" alt="image" src="https://github.com/estuary/connectors/assets/2807772/9ac56dc2-1630-4567-975d-9dbce0f12ddc">

Since credentials is required, it is "open" by default in the form and the user must fill them before they can submit the form, so I think it should be clear for our users that they need to enter their user and password here now. That said, a heads up won't hurt!